src/pages/MyFriendsPage.js

```diff
@@ -0,0 +1,94 @@
+import { Link } from 'react-router-dom';
+import Header from '../components/Header';
+import Button from '../components/Button';
+import FriendCard from '../components/FriendCard';
+import newicon from '../images/new.svg';
+import styled from 'styled-components';
+
+export default function MyFriendsPage({ activities }) {
+  const friends = activities.map(item => {
+    return item.friend;
+  });
+
+  const friendOnlyOnce = friends.filter((friend, index) => {
+    return friends.indexOf(friend) === index;
+  });
+
+  const withoutFriend = friendOnlyOnce.find(
+    friend => friend === 'I still need to plan...'
+  );
+  const onlyFriendsWithName = friendOnlyOnce.filter(
+    friend => friend !== withoutFriend
+  );
+  const sortedFriends = onlyFriendsWithName.sort(function (a, b) {
+    const firstFriend = a.toLowerCase();
+    const secondFriend = b.toLowerCase();
+    if (firstFriend < secondFriend) return -1;
+    if (firstFriend > secondFriend) return 1;
+    return 0;
+  });
+
+  const updatedFriendList = [withoutFriend, ...sortedFriends];
```

Comment:
So..... I went over it and rewrote everything, so that activities without friend are separated from the ones with friends - Thank you for your help yesterday evening 😊

Reply:
Perfect :) you are very welcome.